File: python/twoSum.py
class Solution:
    def twoSum(self, nums, target):
        i = 0
        j = 1
        result = [0, 0]

        explorados = {}
        while i < len(nums):
            if explorados.get(target - nums[i]) != None:
                return [i, explorados.get(target - nums[i])]

            if explorados.get(nums[i]) == None:
                explorados[nums[i]]=i
            
            i += 1
        # while i < len(nums):
        #     while j < len(nums):
        #         if nums[i] + nums[j] == target:
        #             result[0] = i
        #             result[1] = j
        #             return result
        #         j += 1
        #     i += 1
        #     j = i + 1

        return result

File: C++/twoSum.cpp

#include <vector>
using namespace std;

class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        int i = 0;
        int j = 1;
        vector<int> result(2);

        while (i < nums.size()) {
            while (j < nums.size()) {
                if (nums[i] + nums[j] == target) {
                    result[0] = i;
                    result[1] = j;
                    return result;
                }
                j++;
            }
            i++;
            j = i + 1;
        }

        return result;
    }
};